src/php/Compiler/Analyzer/TypeAnalyzer/SpecialForm/LoopSymbol.php

```diff
@@ -68,30 +71,34 @@ public function analyze(Tuple $tuple, NodeEnvironmentInterface $env): LetNode
         }
 
         if (count($lets) > 0) {
-            $bodyExpr = [];
-            for ($i = 2; $i < $tupleCount; $i++) {
-                $bodyExpr[] = $tuple[$i];
-            }
-            $letSym = Symbol::create(Symbol::NAME_LET)->copyLocationFrom($tuple[0]);
-            $letExpr = (Tuple::create($letSym, new Tuple($lets, true), ...$bodyExpr))->copyLocationFrom($tuple);
-            $newExpr = (Tuple::create($tuple[0], new Tuple($preInits, true), $letExpr))->copyLocationFrom($tuple);
+            $bodyExpr = $list->rest()->rest()->toArray();
+            $letSym = Symbol::create(Symbol::NAME_LET)->copyLocationFrom($list->get(0));
```

Comment:
Minor: I saw that we have now `TypeFactory::getInstance()->symbol()` and it's a wrapper for `Symbol::create()`. Actually, it's not being used at all.
The same as `TypeFactory::getInstance()->symbolForNamespace()` also not used.

I still need to think about this concept deeper... 🤔

Reply:
Yes that is still missing. I want to initial new Symbols with the TypeFactory and not with `Symbol::create...`. But maybe we clean this up later.